flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java

```diff
@@ -49,6 +64,22 @@ public EventBus reactor(MDCCleanerThreadPoolExecutor threadPoolExecutor, Environ
                 .env(env)
                 .dispatcher(new ThreadPoolExecutorDispatcher(eventBusThreadPoolBacklogSize, eventBusThreadPoolCoreSize, threadPoolExecutor))
                 .traceEventPath()
+                .dispatchErrorHandler(throwable -> {
+                    if (throwable.getCause() instanceof Exceptions.ValueCause) {
+                        if (((Exceptions.ValueCause) throwable.getCause()).getValue() instanceof Event) {
+                            Event event = (Event) ((Exceptions.ValueCause) throwable.getCause()).getValue();
+                            if (event.getHeaders() != null && event.getHeaders().get("FLOW_ID") != null) {
+                                String flowId = event.getHeaders().get("FLOW_ID").toString();
+                                LOGGER.error("Unhandled exception happened in flow {}, lets cancel it", flowId, throwable);
+                                flowLogDBService.getLastFlowLog(flowId).ifPresent(flowLog -> {
+                                    flowLogDBService.updateLastFlowLogStatus(flowLog, true);
+                                    applicationFlowInformation.handleFlowFail(flowLog);
+                                });
+                            }
+                        }
+                    }
+                    LOGGER.error("Exception happened in dispatcher", throwable);
+                })
```

Comment:
could you move it into a method, so it would be a little bit more readable

Reply:
also there are some call here, maybe we should wrap it with try-catch if something goes wrong during error handling, shouldn't we?

Reply:
Moved to a method and put into try-catch, thanks for the suggestion, makes sense